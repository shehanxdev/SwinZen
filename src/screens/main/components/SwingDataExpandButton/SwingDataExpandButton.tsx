import React, { useState } from 'react';
import { Pressable, View } from 'react-native';
import Collapsible from 'react-native-collapsible';
import LinearGradient from 'react-native-linear-gradient';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextAlignment, TextVariant } from '@sz/constants';

export function SwingDataExpandButton() {
  const [isExpanded, setIsExpanded] = useState(true);

  const gradientConfig = {
    colors: ['#1A5C23', '#16501D', '#16501D', '#0A2F0F'],
    locations: [0.2, 0.7375, 0.7375, 0.241],
    start: { x: 0, y: 1.4 },
    end: { x: 0, y: 0 },
  };

  const swingData = [
    {
      label: 'Club speed',
      value: '95',
    },
    {
      label: 'Ball speed',
      value: '123.5',
    },
    {
      label: 'Smash',
      value: '1.3',
    },
    {
      label: 'Launch',
      value: 'ND',
    },
  ];
  return (
    <View style={tw`relative h-auto pt-10 mb-2`}>
      <Pressable
        style={tw`py-2.5 bg-[${Color.Primary.Sz650}] rounded-2.5 absolute z-1 right-0 left-0`}
        onPress={() => setIsExpanded(prevState => !prevState)}>
        <Text variant={TextVariant.Body2SemiBold} color={Color.Neutral.White}>
          {`Swing data ${isExpanded ? '+' : ' -'}`}
        </Text>
      </Pressable>
      <Collapsible collapsed={isExpanded} style={tw`absolute right-0 left-0 -top-2 `}>
        <LinearGradient
          colors={gradientConfig.colors}
          start={gradientConfig.start}
          end={gradientConfig.end}
          locations={gradientConfig.locations}
          style={tw`pt-6.25 pb-1.75 rounded-2.5`}>
          {swingData.map(data => {
            return (
              <View style={tw`justify-center items-center gap-3.75 flex-row mb-.5`} key={data.label}>
                <View style={tw`flex-1 `}>
                  <Text variant={TextVariant.Body2Regular} color={Color.Neutral.Sz100} textAlign={TextAlignment.Right}>
                    {data.label}
                  </Text>
                </View>
                <View style={tw`flex-1`}>
                  <Text variant={TextVariant.Body1SemiBold} color={Color.Neutral.Sz100} textAlign={TextAlignment.Left}>
                    {data.value}
                  </Text>
                </View>
              </View>
            );
          })}
        </LinearGradient>
      </Collapsible>
    </View>
  );
}
