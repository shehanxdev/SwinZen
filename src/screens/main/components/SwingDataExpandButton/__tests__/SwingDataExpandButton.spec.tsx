import { fireEvent, render } from '@testing-library/react-native';
import * as React from 'react';

import { SwingDataExpandButton } from '../SwingDataExpandButton';

describe('SwingDataExpandButton component', () => {
  const swingData = [
    { label: 'Label 1', value: 'Value 1' },
    { label: 'Label 2', value: 'Value 2' },
  ];
  const testID = 'SwingDataExpandButtonTestID';
  const getRenderedComponent = () => render(<SwingDataExpandButton swingData={swingData} testID={testID} />);

  it(`should render SwingDataExpandButton component correctly`, () => {
    const renderer = getRenderedComponent();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the SwingDataExpandButton component via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });

  it('renders the button with correct text', () => {
    const { getByText } = getRenderedComponent();
    const button = getByText('Swing data +');
    expect(button).toBeTruthy();
  });

  it('expands the content when the button is pressed', () => {
    const { getByText, queryByText } = getRenderedComponent();
    const button = getByText('Swing data +');

    fireEvent.press(button);
    expect(queryByText('Label 1')).toBeTruthy();
    expect(queryByText('Value 1')).toBeTruthy();
    expect(queryByText('Label 2')).toBeTruthy();
    expect(queryByText('Value 2')).toBeTruthy();
  });
});
