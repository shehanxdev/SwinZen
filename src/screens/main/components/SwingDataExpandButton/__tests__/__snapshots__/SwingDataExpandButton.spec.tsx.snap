// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwingDataExpandButton component should render SwingDataExpandButton component correctly 1`] = `
<View
  style={
    {
      "height": "auto",
      "marginBottom": 8,
      "paddingTop": 40,
      "position": "relative",
    }
  }
  testID="SwingDataExpandButtonTestID"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#1A5C23",
        "borderRadius": 10,
        "left": 0,
        "paddingBottom": 10,
        "paddingTop": 10,
        "position": "absolute",
        "right": 0,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "center",
          },
        ]
      }
    >
      Swing data +
    </Text>
  </View>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={false}
      forwardedRef={[Function]}
      onLayout={[Function]}
      style={
        {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": -8,
        }
      }
    >
      <BVLinearGradient
        colors={
          [
            4279917603,
            4279652381,
            4279652381,
            4278857487,
          ]
        }
        endPoint={
          {
            "x": 0,
            "y": 0,
          }
        }
        locations={
          [
            0.2,
            0.7375,
            0.7375,
            0.241,
          ]
        }
        startPoint={
          {
            "x": 0,
            "y": 1.4,
          }
        }
        style={
          {
            "borderRadius": 10,
            "paddingBottom": 7,
            "paddingTop": 25,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 15,
              "justifyContent": "center",
              "marginBottom": 2,
            }
          }
        >
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 24,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "right",
                  },
                ]
              }
            >
              Label 1
            </Text>
          </View>
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Value 1
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 15,
              "justifyContent": "center",
              "marginBottom": 2,
            }
          }
        >
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 24,
                  },
                  {
                    "fontFamily": "SourceSansPro-Regular",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "textAlign": "right",
                  },
                ]
              }
            >
              Label 2
            </Text>
          </View>
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#F8F9FA",
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Value 2
            </Text>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </View>
</View>
`;
